src/OSPSuite.R/Services/ConcurrentSimulationRunner.cs

```diff
@@ -0,0 +1,110 @@
+﻿using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Data;
+using OSPSuite.Core.Domain.Services;
+using OSPSuite.R.Domain;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using SimulationRunOptions = OSPSuite.R.Domain.SimulationRunOptions;
+
+namespace OSPSuite.R.Services
+{
+   public interface IConcurrentSimulationRunner
+   {
+      /// <summary>
+      /// General simulation options
+      /// </summary>
+      SimulationRunOptions SimulationRunOptions { get; set; }
+
+      /// <summary>
+      /// Number of cores to use concurrently. Use 0 or a negative value for using the maximum available.
+      /// </summary>
+      int NumberOfCores { get; set; }
+
+      /// <summary>
+      /// Adds a simulation to the list of Simulations
+      /// </summary>
+      void AddSimulation(IModelCoreSimulation simulation);
+
+      /// <summary>
+      /// Adds a SimulationBatch to the list of SimulationBatches
+      /// </summary>
+      /// <param name="simulationBatch"></param>
+      void AddSimulationBatch(SimulationBatch simulationBatch);
+
+      void Clear();
+
+      /// <summary>
+      /// Runs all preset settings concurrently
+      /// </summary>
+      /// <returns></returns>
+      SimulationResults[] RunConcurrently();
+   }
+
+   public class ConcurrentSimulationRunner : IConcurrentSimulationRunner
```

Comment:
I would make this guy umplement IDisposable and ensure that Dispose called Clear if not done already

This is the snippet we use

```

      private bool _disposed;

      public void Dispose()
      {
         if (_disposed) return;

         Cleanup();
         GC.SuppressFinalize(this);
         _disposed = true;
      }

      ~ConcurrentSimulationRunner ()
      {
         Cleanup();
      }

      protected virtual void Cleanup()
      {
      }
```